src/main/java/no/sikt/nva/scrapers/DublinCoreValidator.java

```diff
@@ -129,17 +132,37 @@ private static Optional<WarningDetails> getPageNumberWarning(DublinCore dublinCo
     }
 
     private static Optional<ErrorDetails> getDoiErrorDetails(DublinCore dublinCore) {
-        var doiErrors =
+        var doiList =
             dublinCore.getDcValues()
                 .stream()
-                .filter(DublinCoreValidator::hasInvalidDoi)
+                .filter(DcValue::isDoi)
                 .map(DcValue::getValue)
-                .collect(
-                    Collectors.toList());
-        if (doiErrors.isEmpty()) {
+                .map(DublinCoreValidator::addHttpStringIfNotPresent)
+                .collect(Collectors.toList());
+
+        return validateDoiList(doiList);
+    }
+
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    @NotNull
+    private static Optional<ErrorDetails> validateDoiList(List<String> doiList) {
+        if (doiList.isEmpty()) {
             return Optional.empty();
         } else {
-            return Optional.of(new ErrorDetails(Error.INVALID_DOI, doiErrors));
+            for (String doi : doiList) {
+                if (!isValidDoiOffline(doi)) {
+                    return Optional.of(new ErrorDetails(Error.INVALID_DOI_OFFLINE_CHECK, doiList));
+                }
+            }
+        }
+        return Optional.empty();
+    }
```

Comment:
Denne metoden er merkelig. Fra koden er det veldig utydelig om en ressurs kan ha én eller flere DOI'er. For min del ser det ut som den returnerer Optional.empty hvis den ikke finner noen invalide DOI'er, men så er den merkelig fordi hvis det er to DOI'er som er invalide så vil den kun returnere den første den støter på som invalid hvis jeg forstår denne koden riktig.

Forøvrig denne kode snutten ser mistenkelig lik en annen kode snutt.

Reply:
ordnet